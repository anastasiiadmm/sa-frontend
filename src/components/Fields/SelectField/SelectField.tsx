import { Select } from 'antd';
import React from 'react';

import { climateOptions } from 'types/types';

interface Props {
  handleChange?: () => void;
  defaultValue?: string;
  disabled?: boolean;
  customStyle?: string;
  options?: climateOptions[];
}

const SelectField: React.FC<Props> = ({
  handleChange,
  disabled,
  defaultValue,
  options,
  customStyle,
}) => {
  return (
    <Select
      disabled={disabled}
      defaultValue={defaultValue}
      style={{ width: customStyle ?? 'auto' }}
      onChange={handleChange}
      options={options}
    />
  );
};

export default SelectField;
