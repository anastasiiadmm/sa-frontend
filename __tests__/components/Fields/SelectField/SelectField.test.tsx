import { render, cleanup } from '@testing-library/react';
import "@testing-library/jest-dom";

import SelectField from "../../../../src/components/Fields/SelectField/SelectField";
import * as TestRenderer from "react-test-renderer";

jest.mock('antd', () => ({
    Select: ({ children, ...props }:any) => <select {...props}>{children}</select>,
}));

const climateOptionsMock = [
    { value: 'Tropical', label: 'Tropical' },
    { value: 'Continental', label: 'Continental' },
];

afterEach(cleanup);

describe('SelectField Component', () => {
    it('renders without crashing', () => {
        render(<SelectField options={climateOptionsMock} />);
    });

    it('should render with default value', () => {
        const tree = TestRenderer.create(
            <SelectField defaultValue="Continental" options={climateOptionsMock} />
        );
        expect(tree).toMatchSnapshot();
    });

    it('', () => {
        const { getByRole } = render(
            <SelectField disabled options={climateOptionsMock} />
        );
        expect(getByRole('combobox')).toBeDisabled();
    });
});
