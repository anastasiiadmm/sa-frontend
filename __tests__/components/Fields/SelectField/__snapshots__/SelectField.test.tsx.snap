// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SelectField Component should render with default value 1`] = `
<div
  className="ant-select css-dev-only-do-not-override-1vr7spz ant-select-single ant-select-show-arrow"
  onBlur={[Function]}
  onFocus={[Function]}
  onKeyDown={[Function]}
  onKeyUp={[Function]}
  onMouseDown={[Function]}
  style={
    {
      "width": "220px",
    }
  }
>
  <div
    className="ant-select-selector"
    onClick={[Function]}
    onMouseDown={[Function]}
  >
    <span
      className="ant-select-selection-search"
    >
      <input
        aria-autocomplete="list"
        aria-controls="undefined_list"
        aria-expanded={false}
        aria-haspopup="listbox"
        aria-label="Search"
        aria-owns="undefined_list"
        autoComplete="off"
        className="ant-select-selection-search-input"
        disabled={false}
        onChange={[Function]}
        onCompositionEnd={[Function]}
        onCompositionStart={[Function]}
        onKeyDown={[Function]}
        onMouseDown={[Function]}
        onPaste={[Function]}
        readOnly={true}
        role="combobox"
        style={
          {
            "opacity": 0,
          }
        }
        type="search"
        unselectable="on"
        value=""
      />
    </span>
    <span
      className="ant-select-selection-item"
      title="Continental"
    >
      Continental
    </span>
  </div>
  <span
    aria-hidden={true}
    className="ant-select-arrow"
    onMouseDown={[Function]}
    style={
      {
        "WebkitUserSelect": "none",
        "userSelect": "none",
      }
    }
    unselectable="on"
  >
    <img
      alt="arrowDown"
      src="test.jpg"
    />
  </span>
</div>
`;
